agent/internal/task_config.go

```diff
@@ -84,6 +86,11 @@ type CommandCleanup struct {
 	Run func(context.Context) error
 }
 
+type AssumeRoleInformation struct {
+	RoleARN    string
+	Expiration string
```

Comment:
nit: rather than lazily parsing the string each time we want to check it, would it be simpler to eagerly parse the string when adding it to the assume role cache?